### PR TITLE
[rubysrc2cpg] Simplified Test Fixtures & Added Class Init Method

### PR DESCRIPTION
**Pull request by DavidBakerEffendi**
* It is invalid to have a body directly under a type declaration in terms of using the `Node.method` implementation, but also logically we should use the `<clinit>`-style method. So this has been implemented
* Also fixed the ruby test fixture to not attempt to run duplicate overlays and have combined the test fixtures into one, configurable test fixture.

CC: @khemrajrathore 

Resolves https://github.com/joernio/joern/issues/3064

### CODE REVIEW

**Review by xavierpinho — APPROVED** [review]
Resolves #3064

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/testfixtures/RubyCode2CpgFixture.scala [hunk 1]
```diff
@@ -17,24 +19,34 @@ trait RubyFrontend extends LanguageFrontend {
...
+  override def applyPasses(): Unit = {
+    X2Cpg.applyDefaultOverlays(this)
+
+    if (withDataFlow) {
+      val context = new LayerCreatorContext(this)
+      val options = new OssDataFlowOptions()
+      new OssDataFlow(options).run(context)
+    }
+
+    if (withPostProcessing) {
```
Comment (khemrajrathore) [inline joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/testfixtures/RubyCode2CpgFixture.scala:39]: Shouldn't the post-processing pass be applied before Dataflow pass?
